ibis/sql/postgres/tests/test_udf.py

```diff
@@ -0,0 +1,225 @@
+    con.con.execute(
+        "CREATE SCHEMA IF NOT EXISTS {};".format(schema_name)
+    )
+    return schema_name
+
+
+@pytest.fixture(scope='session')
+def table_name():
+    return 'udf_test_users'
+
+
+@pytest.fixture(scope='session')
+def sql_table_setup(test_schema, table_name):
+    return """DROP TABLE IF EXISTS {schema}.{table_name};
+CREATE TABLE {schema}.{table_name} (
+    user_id integer,
+    user_name varchar,
+    name_length integer
+);
+INSERT INTO {schema}.{table_name} VALUES
+(1, 'Raj', 3),
+(2, 'Judy', 4),
+(3, 'Jonathan', 8)
+;
+""".format(schema=test_schema, table_name=table_name)
+
+
+@pytest.fixture(scope='session')
+def sql_define_py_udf(test_schema):
+    return """CREATE OR REPLACE FUNCTION {schema}.pylen(x varchar)
+RETURNS integer
+LANGUAGE plpythonu
+AS
+$$
+return len(x)
+$$;""".format(schema=test_schema)
+
+
+@pytest.fixture(scope='session')
+def sql_define_udf(test_schema):
+    return """CREATE OR REPLACE FUNCTION {schema}.custom_len(x varchar)
+RETURNS integer
+LANGUAGE SQL
+AS
+$$
+SELECT length(x);
+$$;""".format(schema=test_schema)
+
+
+@pytest.fixture(scope='session')
+def con_for_udf(
+        con,
+        test_schema,
+        sql_table_setup,
+        sql_define_udf,
+        sql_define_py_udf
+):
+    con.con.execute(sql_table_setup)
+    con.con.execute(sql_define_udf)
+    con.con.execute(sql_define_py_udf)
+    try:
+        yield con
+    finally:
+        # teardown
+        con.con.execute("DROP SCHEMA IF EXISTS {} CASCADE".format(test_schema))
+
+
+@pytest.fixture
+def table(con_for_udf, table_name, test_schema):
+    return con_for_udf.table(table_name, schema=test_schema)
+
+# Tests
+
+
+def test_sql_length_udf_worked(test_schema, table):
+    """Test creating ibis UDF object based on existing UDF in the database"""
+    # Create ibis UDF objects referring to UDFs already created in the database
+    custom_length_udf = existing_udf(
+        'custom_len',
+        input_types=[dt.string],
+        output_type=dt.int32,
+        schema=test_schema
+    )
+    result_obj = table[
+        table,
+        custom_length_udf(table['user_name']).name('custom_len')
+    ]
+    result = result_obj.execute()
+    assert result['custom_len'].sum() == result['name_length'].sum()
+
+
+def test_py_length_udf_worked(test_schema, table):
+    # Create ibis UDF objects referring to UDFs already created in the database
+    py_length_udf = existing_udf(
+        'pylen',
+        input_types=[dt.string],
+        output_type=dt.int32,
+        schema=test_schema
+    )
+    result_obj = table[
+        table,
+        py_length_udf(table['user_name']).name('custom_len')
+    ]
+    result = result_obj.execute()
+    assert result['custom_len'].sum() == result['name_length'].sum()
+
+
+def mult_a_b(a, b):
+    """Test function to be defined in-database as a UDF
+    and used via ibis UDF"""
+    return a * b
+
+
+def test_func_to_udf_smoke(con_for_udf, test_schema, table):
```

Comment:
Took off `_smoke`. (I originally called it a smoke test because it tests end-to-end functionality including going through the backend, but most of the tests are doing that, so you're right, it doesn't make sense to call it a smoke test.)